src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/EditorConfig/Parsing/EditorConfigParser.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.EditorConfig.Parsing
+{
+    internal static class EditorConfigParser
+    {
+        // Matches EditorConfig section header such as "[*.{js,py}]", see https://editorconfig.org for details
+        private static readonly Regex s_sectionMatcher = new(@"^\s*\[(([^#;]|\\#|\\;)+)\]\s*([#;].*)?$", RegexOptions.Compiled);
+        // Matches EditorConfig property such as "indent_style = space", see https://editorconfig.org for details
+        private static readonly Regex s_propertyMatcher = new(@"^\s*([\w\.\-_]+)\s*[=:]\s*(.*?)\s*([#;].*)?$", RegexOptions.Compiled);
```

Comment:
@jmarolf One of the reasons I'm worried about two parser implementations is something like https://github.com/dotnet/roslyn/pull/51625, which proposes a change to the `s_propertyMatcher` regex. It's likely we can miss any changes of those kind here. Probably some code comments in each place referring to the other would do it for now until this is unified?

Reply:
in a world where there was a formal spec that we could write against this would be less of a concern but things being as they are I agree we don't want to do this for no reason. I am still of the opinion that this is justified considering the compilers' requirements.

I will add test cases for the comment parsing to both implementations so that if someone ever updates one version and not the other CI will fail.